### PR TITLE
docs(rxSearchBox): FRMW-191 Migrate rxSearchBox

### PR DESCRIPTION
**Pull request by ssalinas24**
JIRA: https://jira.rax.io/browse/FRMW-191

- [x] Dev LGTM
- [x] Dev LGTM

### CODE REVIEW

File: src/rxSearchBox/rxSearchBox.js [hunk 1]
```diff
@@ -18,8 +18,21 @@ angular.module('encore.ui.rxSearchBox', [])
  * The rxSearchBox directive behaves similar to the HTML "Search" input type. When
  * the search box is not empty, an "X" button within the element will allow you to
  * clear the value. Once clear, the "X" will disappear. A disabled search box cannot
- * be cleared of its value via the "X" button because the button will display.
+ * be cleared of its value via the "X" button because the button will not display.
  *
+ * Though it is described as a search box, you can also use it for filtering capabilities
+ * (as seen by the placeholder text in the "Customized" demo).
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: Please make "demo" a link to the component demo.

```
(as seen by the placeholder text in the "Customized" {@link /#/components/rxSearchBox demo})
```

**Comment by ssalinas24** [comment]
@CITguy I have made the update for the link in my second commit.

**Comment by CITguy** [comment]
LGTM,  @ssalinas24 please squash in preparation for merge.

**Comment by ssalinas24** [comment]
@CITguy the commits have been squashed.

**Comment by bry4n** [comment]
LGTM

File: src/rxSearchBox/rxSearchBox.js [hunk 2]
```diff
@@ -18,8 +18,21 @@ angular.module('encore.ui.rxSearchBox', [])
...
  * clear the value. Once clear, the "X" will disappear. A disabled search box cannot
- * be cleared of its value via the "X" button because the button will display.
+ * be cleared of its value via the "X" button because the button will not display.
  *
+ * Though it is described as a search box, you can also use it for filtering capabilities
+ * (as seen by the placeholder text in the "Customized" {@link /#/components/rxSearchBox demo}).
+ *
+ * # Styling
+ * You can style the `<rx-search-box>` element via custom CSS classes the same way you would any HTML element.
+ * See the customized search box in the demo for an example.
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: Sorry, overlooked this.

Please make "demo" a link to the component page (same as above).

File: src/rxSearchBox/rxSearchBox.js [hunk 3]
```diff
@@ -18,8 +18,21 @@ angular.module('encore.ui.rxSearchBox', [])
...
+ *
+ * <pre>
+ * <rx-search-box
+ *      class="wide-search-box"
```
Comment (CITguy) [inline src/rxSearchBox/rxSearchBox.js]: `class` is unnecessary, please remove

**Comment by CITguy** [comment]
Sorry, two small things to address before we merge this one.

File: src/rxSearchBox/rxSearchBox.js (hunk 3 above)
Comment (ssalinas24) [inline src/rxSearchBox/rxSearchBox.js]: I added this example to demonstrate the custom class. If removing, should i remove line 32 entirely, or just get rid of the entire example?
NOTE: A second example already exists on the page to demonstrate the general rx-search-box.

File: src/rxSearchBox/rxSearchBox.js [hunk 4]
```diff
@@ -15,19 +16,37 @@ angular.module('encore.ui.rxSearchBox', [])
...
+ * {@link /#/components/rxSearchBox demo}).
+ *
+ * # Styling
+ * You can style the `<rx-search-box>` element via custom CSS classes the same
+ * way you would any HTML element. See the customized search box in the
+ * {@link /#/components/rxSearchBox demo} for an example.
+ *
+ * <pre>
+ * <rx-search-box
+ *      ng-model="customSearchModel"
```
Comment (ssalinas24) [inline src/rxSearchBox/rxSearchBox.js]: @CITguy I went ahead and removed the entire Class="Wide-..." line and kept the example. I also added the second demo link.

**Pull request merged**